## Part 1: Explore the Data

Import the data and use Pandas to learn more about the dataset.

In [2]:
import pandas as pd

df = pd.read_csv('Resources/client_dataset.csv')
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year         item_id     category         subcategory  \
0          17        2023  EUD29711-63-6U        decor            wall art   
1          20        2023  XDA18116-89-4A  consumables                pens   
2           4        2023  ABE59463-05-7E     software  project management   
3          17        2023  ZMM00836-65-0C  consumables                pens   
4          21        2023  BZX55559-12-3X  consumables                misc   

   unit_price  unit_cost  unit_weight  qty  line_number  
0     1096.80     762.71         7.50  105            1  
1       24.95      15.09         1.49   21            0  
2       13.52       7.86         1.68   39            6  
3       36.42      24.85         1.23   29            3  
4      195.10     108.17        46.43   20            1

In [3]:
# View the column names in the data
df.columns

Index(['first', 'last', 'job', 'phone', 'email', 'client_id', 'order_id',
       'order_date', 'order_week', 'order_year', 'item_id', 'category',
       'subcategory', 'unit_price', 'unit_cost', 'unit_weight', 'qty',
       'line_number'],
      dtype='object')

In [4]:
# Use the describe function to gather some basic statistics
df.describe()

client_id      order_id    order_week    order_year    unit_price  \
count  54639.000000  5.463900e+04  54639.000000  54639.000000  54639.000000   
mean   54837.869416  5.470190e+06     11.359139   2022.993064    136.267207   
std    25487.438231  2.599807e+06      7.023499      0.082997    183.873135   
min    10033.000000  1.000886e+06      1.000000   2022.000000      0.010000   
25%    33593.000000  3.196372e+06      6.000000   2023.000000     20.800000   
50%    53305.000000  5.496966e+06     11.000000   2023.000000     68.310000   
75%    78498.000000  7.733869e+06     17.000000   2023.000000    173.160000   
max    99984.000000  9.998480e+06     52.000000   2023.000000   1396.230000   

          unit_cost   unit_weight           qty   line_number  
count  54639.000000  54639.000000  5.463900e+04  54639.000000  
mean      99.446073      5.004116  5.702646e+02      2.979667  
std      133.164267      5.326599  1.879552e+04      2.436320  
min        0.010000      0.000000  0.000000e+00      0.000000  
25%       14.840000      1.450000  3.200000e+01      1.000000  
50%       49.890000      3.240000  6.800000e+01      3.000000  
75%      125.570000      6.890000  1.700000e+02      5.000000  
max      846.270000     46.430000  3.958244e+06      9.000000

In [5]:
# Use this space to do any additional research
# and familiarize yourself with the data.
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54639 entries, 0 to 54638
Data columns (total 18 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   first        54639 non-null  object 
 1   last         54639 non-null  object 
 2   job          54639 non-null  object 
 3   phone        54639 non-null  object 
 4   email        54639 non-null  object 
 5   client_id    54639 non-null  int64  
 6   order_id     54639 non-null  int64  
 7   order_date   54639 non-null  object 
 8   order_week   54639 non-null  int64  
 9   order_year   54639 non-null  int64  
 10  item_id      54639 non-null  object 
 11  category     54639 non-null  object 
 12  subcategory  54639 non-null  object 
 13  unit_price   54639 non-null  float64
 14  unit_cost    54639 non-null  float64
 15  unit_weight  54639 non-null  float64
 16  qty          54639 non-null  int64  
 17  line_number  54639 non-null  int64  
dtypes: float64(3), int64(6), object(9)
memory usag

In [6]:
# What three item categories had the most entries?
df['category'].value_counts().nlargest(3)

category
consumables    23538
furniture      11915
software        8400
Name: count, dtype: int64

In [7]:
df['category']

0              decor
1        consumables
2           software
3        consumables
4        consumables
            ...     
54634       software
54635    consumables
54636    consumables
54637      furniture
54638       software
Name: category, Length: 54639, dtype: object

In [8]:
# For the category with the most entries, which subcategory had the most entries?
df['subcategory'].value_counts().nlargest(1)

subcategory
bathroom supplies    6424
Name: count, dtype: int64

In [9]:
# Which five clients had the most entries in the data?
df['client_id'].value_counts().nlargest()

client_id
33615    220
66037    211
46820    209
38378    207
24741    207
Name: count, dtype: int64

In [10]:
# Store the client ids of those top 5 clients in a list.
most_entries = df['client_id'].value_counts().nlargest().index.tolist()
most_entries

[33615, 66037, 46820, 38378, 24741]

In [11]:
# How many total units (the qty column) did the client with the most entries order order?
total_units = df.loc[df['client_id'] == most_entries[0]]['qty'].sum()
total_units

64313

## Part 2: Transform the Data
Do we know that this client spent the more money than client 66037? If not, how would we find out? Transform the data using the steps below to prepare it for analysis.

In [13]:
# Create a column that calculates the subtotal for each line using the unit_price and the qty
df['line_subtotal'] = df['unit_price'] * df['qty']
df

first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id  order_id  order_date  \
0         harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1        myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                    swatson8146@payne.net      57113   1902144  2023-01-29   
3                   bakernathan@benson.com      46554   9031802  2023-04-25   
4      christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   
...                                    ...        ...       ...         ...   
54634   late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635         derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636            gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637            solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                     belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year         item_id     category  \
0              17        2023  EUD29711-63-6U        decor   
1              20        2023  XDA18116-89-4A  consumables   
2               4        2023  ABE59463-05-7E     software   
3              17        2023  ZMM00836-65-0C  consumables   
4              21        2023  BZX55559-12-3X  consumables   
...           ...         ...             ...          ...   
54634          17        2023  UCA28532-31-0B     software   
54635           6        2023  EBD29277-23-7U  consumables   
54636           9        2023  EBD30093-55-9Z  consumables   
54637           3        2023  XMZ38239-49-3D    furniture   
54638           1        2023  ZCZ05589-50-5D     software   

              subcategory  unit_price  unit_cost  unit_weight   qty  \
0                wall art     1096.80     762.71         7.50   105   
1                    pens       24.95      15.09         1.49    21   
2      project management       13.52       7.86         1.68    39   
3                    pens       36.42      24.85         1.23    29   
4                    misc      195.10     108.17        46.43    20   
...                   ...         ...        ...          ...   ...   
54634           logistics       83.13      51.60         2.25    33   
54635    kitchen supplies      206.59     175.46        11.70    47   
54636   bathroom supplies       65.66      57.31         4.16   475   
54637              stands        1.48       1.22        18.04   112   
54638  project management        3.01       1.61         2.07  1031   

       line_number  line_subtotal  
0                1      115164.00  
1                0         523.95  
2                6         527.28  
3                3        1056.18  
4                1        3902.00  
...            ...            ...  
54634            8        2743.29  
54635            0        9709.73  
54636            6       31188.50  
54637            0         165.76  
54638            5        3103.31  

[54639 rows x 19 columns]

In [14]:
df['total_weight'] = df['qty'] * df['unit_weight']

In [15]:
# Create a column for shipping price.
# Assume a shipping price of $7 per pound for orders over 50 pounds and $10 per pound for items 50 pounds or under.
def shipping_price(weight_in_lbs):
    if weight_in_lbs > 50:
        return 7 * weight_in_lbs 
    return 10 * weight_in_lbs 
df['shipping_price'] = df['total_weight'].apply(shipping_price)
df.head()

first      last                    job               phone  \
0     Donald   Harding           Immunologist  793-904-7725x39308   
1    Tiffany     Myers        Music therapist    201.442.4543x942   
2    Shannon    Watson           Immunologist   687.737.9424x8503   
3     Nathan     Baker  Accounting technician    827-788-8123x012   
4  Christina  Schwartz           Chiropractor        265-829-3643   

                                 email  client_id  order_id  order_date  \
0     harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1    myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                swatson8146@payne.net      57113   1902144  2023-01-29   
3               bakernathan@benson.com      46554   9031802  2023-04-25   
4  christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   

   order_week  order_year  ...     category         subcategory unit_price  \
0          17        2023  ...        decor            wall art    1096.80   
1          20        2023  ...  consumables                pens      24.95   
2           4        2023  ...     software  project management      13.52   
3          17        2023  ...  consumables                pens      36.42   
4          21        2023  ...  consumables                misc     195.10   

   unit_cost  unit_weight  qty  line_number  line_subtotal  total_weight  \
0     762.71         7.50  105            1      115164.00        787.50   
1      15.09         1.49   21            0         523.95         31.29   
2       7.86         1.68   39            6         527.28         65.52   
3      24.85         1.23   29            3        1056.18         35.67   
4     108.17        46.43   20            1        3902.00        928.60   

   shipping_price  
0         5512.50  
1          312.90  
2          458.64  
3          356.70  
4         6500.20  

[5 rows x 21 columns]

In [16]:
# Create a column for the total price using the subtotal and the shipping price along with a sales tax of 9.25%
df['total_price'] = ((df['shipping_price'] + df['line_subtotal']) * 1.0925).round(2)
df

first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id  order_id  order_date  \
0         harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1        myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                    swatson8146@payne.net      57113   1902144  2023-01-29   
3                   bakernathan@benson.com      46554   9031802  2023-04-25   
4      christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   
...                                    ...        ...       ...         ...   
54634   late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635         derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636            gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637            solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                     belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year  ...         subcategory unit_price unit_cost  \
0              17        2023  ...            wall art    1096.80    762.71   
1              20        2023  ...                pens      24.95     15.09   
2               4        2023  ...  project management      13.52      7.86   
3              17        2023  ...                pens      36.42     24.85   
4              21        2023  ...                misc     195.10    108.17   
...           ...         ...  ...                 ...        ...       ...   
54634          17        2023  ...           logistics      83.13     51.60   
54635           6        2023  ...    kitchen supplies     206.59    175.46   
54636           9        2023  ...   bathroom supplies      65.66     57.31   
54637           3        2023  ...              stands       1.48      1.22   
54638           1        2023  ...  project management       3.01      1.61   

       unit_weight   qty  line_number  line_subtotal  total_weight  \
0             7.50   105            1      115164.00        787.50   
1             1.49    21            0         523.95         31.29   
2             1.68    39            6         527.28         65.52   
3             1.23    29            3        1056.18         35.67   
4            46.43    20            1        3902.00        928.60   
...            ...   ...          ...            ...           ...   
54634         2.25    33            8        2743.29         74.25   
54635        11.70    47            0        9709.73        549.90   
54636         4.16   475            6       31188.50       1976.00   
54637        18.04   112            0         165.76       2020.48   
54638         2.07  1031            5        3103.31       2134.17   

       shipping_price  total_price  
0             5512.50    131839.08  
1              312.90       914.26  
2              458.64      1077.12  
3              356.70      1543.57  
4             6500.20     11364.40  
...               ...          ...  
54634          519.75      3564.87  
54635         3849.30     14813.24  
54636        13832.00     

In [17]:
# Create a column for the cost of each line using unit cost, qty, and
# shipping price (assume the shipping cost is exactly what is charged to the client).
df['total_cost'] = (df['unit_cost'] * df['qty']) + df['shipping_price']
df

first       last                      job               phone  \
0         Donald    Harding             Immunologist  793-904-7725x39308   
1        Tiffany      Myers          Music therapist    201.442.4543x942   
2        Shannon     Watson             Immunologist   687.737.9424x8503   
3         Nathan      Baker    Accounting technician    827-788-8123x012   
4      Christina   Schwartz             Chiropractor        265-829-3643   
...          ...        ...                      ...                 ...   
54634     Lauren      Reese           Radio producer  977.877.5272x11382   
54635    Derrick      Moore                 Musician        358.661.5483   
54636     Monica  Gutierrez         Graphic designer    294.805.9100x339   
54637      Wanda    Solomon             Toxicologist       (311)767-4924   
54638     Dwayne       Bell  Chief Financial Officer   606-619-7667x9508   

                                     email  client_id  order_id  order_date  \
0         harding.donald.7185@sullivan.com      58515   8953482  2023-04-28   
1        myers.t.6537@ferguson-johnson.net      37609   8069089  2023-05-19   
2                    swatson8146@payne.net      57113   1902144  2023-01-29   
3                   bakernathan@benson.com      46554   9031802  2023-04-25   
4      christinaschwartz9252@mcconnell.com      92089   1322274  2023-05-28   
...                                    ...        ...       ...         ...   
54634   late_reese_4081@montoya-chavez.org      29043   9021716  2023-04-26   
54635         derrick.moore.2602@pope.info      41908   6290153  2023-02-08   
54636            gutierrezm3195@morris.org      35176   8692622  2023-03-05   
54637            solomonwanda5962@ross.org      24485   7592730  2023-01-18   
54638                     belld@galvan.com      24007   7489403  2023-01-04   

       order_week  order_year  ... unit_price unit_cost unit_weight   qty  \
0              17        2023  ...    1096.80    762.71        7.50   105   
1              20        2023  ...      24.95     15.09        1.49    21   
2               4        2023  ...      13.52      7.86        1.68    39   
3              17        2023  ...      36.42     24.85        1.23    29   
4              21        2023  ...     195.10    108.17       46.43    20   
...           ...         ...  ...        ...       ...         ...   ...   
54634          17        2023  ...      83.13     51.60        2.25    33   
54635           6        2023  ...     206.59    175.46       11.70    47   
54636           9        2023  ...      65.66     57.31        4.16   475   
54637           3        2023  ...       1.48      1.22       18.04   112   
54638           1        2023  ...       3.01      1.61        2.07  1031   

       line_number  line_subtotal  total_weight  shipping_price  total_price  \
0                1      115164.00        787.50         5512.50    131839.08   
1                0         523.95         31.29          312.90       914.26   
2                6         527.28         65.52          458.64      1077.12   
3                3        1056.18         35.67          356.70      1543.57   
4                1        3902.00        928.60         6500.20     11364.40   
...            ...            ...           ...             ...          ...   
54634            8        2743.29         74.25          519.75      3564.87   
54635            0        9709.73        549.90         3849.30     14813.24   
54636            6       31188.50       1976.00        13832.00     49184.90   
54637            0         165.76       2020.48        14143.36     15632.71   
54638            5        3103.31       2134.17        14939.19     19711.43   

       total_cost  
0        85597.05  
1          629.79  
2          765.18  
3         1077.35  
4         8663.60  
...           ...  
54634     2222.55  
54635    12095.92  
54636    41054.25  
54637    14280.00  
54638    16599.10  

[54639 rows x 23 colum

In [18]:
# Create a column for the profit of each line using line cost and line price
df['profit'] = df['total_price'] - df['total_cost']
df.head(50)

first        last                                           job  \
0        Donald     Harding                                  Immunologist   
1       Tiffany       Myers                               Music therapist   
2       Shannon      Watson                                  Immunologist   
3        Nathan       Baker                         Accounting technician   
4     Christina    Schwartz                                  Chiropractor   
5       Jessica       Reyes  Production designer, theatre/television/film   
6         Brian       Moore                                       Barista   
7         Jesus      Wright                            Theme park manager   
8       Rebecca       Brown                                   Chiropodist   
9          Greg      Garcia                            Secretary, company   
10         Kyle    Williams           Engineer, broadcasting (operations)   
11      Roberto      Howard                      Surveyor, land/geomatics   
12      Jessica       Reyes  Production designer, theatre/television/film   
13     Patricia       Gould                                      Lobbyist   
14    Katherine      Taylor                        Psychologist, clinical   
15      Matthew      Orozco                    Research scientist (maths)   
16       Joseph      Jordan            Government social research officer   
17       Carrie       Green                           Colour technologist   
18        David       Reyes              Production assistant, television   
19       Shelby    Williams                           Colour technologist   
20       Tricia  Stephenson                   Lecturer, further education   
21      Patrick    Anderson                Commercial art gallery manager   
22        Peter     Roberts                    Architectural technologist   
23  Christopher      Rhodes               Engineer, manufacturing systems   
24        Glenn      Garcia                              Textile designer   
25        Peter     Roberts                    Architectural technologist   
26      Patrick    Anderson                Commercial art gallery manager   
27        Jacob     Mendoza                                Air cabin crew   
28      Tiffany        Lowe                           Medical illustrator   
29        Caleb        Rice                              Technical brewer   
30      Deborah        Cook                     Radiographer, therapeutic   
31      Richard     Kennedy                        Race relations officer   
32       Connie      Harris                       Horticulturist, amenity   
33      Cynthia      Palmer                      Surveyor, rural practice   
34         Erin   Robertson                                Science writer   
35    Catherine      Barnes                                 Airline pilot   
36        Corey       Wells             Environmental health practitioner   
37      Bradley     Spencer                             Warehouse manager   
38       Andrea      Brooks                  IT technical support officer   
39      Mallory    Mitchell                                Quarry manager   
40      William      Mccann                         Accountant, chartered   
41      Douglas        Bean                      Doctor, general practice   
42       Victor    Griffith                           Structural engineer   
43       Taylor      Wagner                                  Toxicologist   
44        Barry     Johnson                     Education officer, museum   
45       Daniel      Newman                    Designer, interior/spatial   
46       Ashley   Hernandez                                  Retail buyer   
47      Vincent       Davis                  Designer, exhibition/display   
48       Daniel      Newman                    Designer, interior/spatial   
49       Curtis      Garcia                                    Bookseller   

                     phone                                     email  \
0       79

## Part 3: Confirm your work
You have email receipts showing that the total prices for 3 orders. Confirm that your calculations match the receipts. Remember, each order has multiple lines.

Order ID 2742071 had a total price of \$152,811.89

Order ID 2173913 had a total price of \$162,388.71

Order ID 6128929 had a total price of \$923,441.25


In [20]:
# Check your work using the totals above
orders = [2742071, 2173913, 6128929]
for i in orders:
    print(f"Order ID {i} had a total price of ${df['total_price'].loc[df['order_id'] == i].sum().round(2)}")

Order ID 2742071 had a total price of $152811.89
Order ID 2173913 had a total price of $162388.71
Order ID 6128929 had a total price of $923441.25


## Part 4: Summarize and Analyze
Use the new columns with confirmed values to find the following information.

In [22]:
# How much did each of the top 5 clients by quantity spend? Check your work from Part 1 for client ids.
top_5 = df['client_id'].value_counts().nlargest().index
grouped_df = df.groupby('client_id')['total_price'].sum()
for i in top_5:
    print(f"Client {i} spent ${grouped_df[i].round(2)}")

Client 33615 spent $8377308.52
Client 66037 spent $10259514.79
Client 46820 spent $9743794.36
Client 38378 spent $12906550.87
Client 24741 spent $82268892.02


In [23]:
# Create a summary DataFrame showing the totals for the for the top 5 clients with the following information:
# total units purchased, total shipping price, total revenue, and total profit. 
grouped_df = df.groupby('client_id').sum()
cleaned_grouped_df = grouped_df[['qty', 'shipping_price', 'total_price', 'total_cost', 'profit']]
top_5_df = cleaned_grouped_df.loc[top_5]
top_5_df

qty  shipping_price  total_price   total_cost       profit
client_id                                                               
33615       64313      1828984.89   8377308.52   6175313.91   2201994.61
66037       43018      1395151.85  10259514.79   7004482.98   3255031.81
46820       75768      1601448.84   9743794.36   7007191.64   2736602.72
38378       73667      3429455.40  12906550.87   9634720.98   3271829.89
24741      239862      5126448.37  82268892.02  45688899.71  36579992.31

In [24]:
# Format the data and rename the columns to names suitable for presentation.
top_5_df = top_5_df.rename_axis('Client ID')
top_5_df = top_5_df.rename(columns={'qty':'Units', 
                                    'shipping_price':'Shipping(millions)',
                                    'total_price':'Total Revenue(millions)',
                                    'total_cost':'Total Cost(millions)',
                                    'profit':'Total Profit(millions)'})
# Define the money columns. 
money_cols = ['Shipping(millions)', 'Total Revenue(millions)', 'Total Cost(millions)', 'Total Profit(millions)']

# Define a function that converts a dollar amount to millions.
def dollar_to_mil(x):
    new_val = x/1000000
    return new_val
    
# Apply the currency_format_millions function to only the money columns. 
top_5_df[money_cols] = top_5_df[money_cols].apply(dollar_to_mil)

# Rename the columns to reflect the change in the money format. 
top_5_df



Units  Shipping(millions)  Total Revenue(millions)  \
Client ID                                                        
33615       64313            1.828985                 8.377309   
66037       43018            1.395152                10.259515   
46820       75768            1.601449                 9.743794   
38378       73667            3.429455                12.906551   
24741      239862            5.126448                82.268892   

           Total Cost(millions)  Total Profit(millions)  
Client ID                                                
33615                  6.175314                2.201995  
66037                  7.004483                3.255032  
46820                  7.007192                2.736603  
38378                  9.634721                3.271830  
24741                 45.688900               36.579992

In [48]:
# Sort the updated data by "Total Profit (millions)" form highest to lowest and assign the sort to a new DatFrame.
sorted_top_5_df = top_5_df.sort_values(by='Total Profit(millions)', ascending=False)
sorted_top_5_df

Units  Shipping(millions)  Total Revenue(millions)  \
Client ID                                                        
24741      239862            5.126448                82.268892   
38378       73667            3.429455                12.906551   
66037       43018            1.395152                10.259515   
46820       75768            1.601449                 9.743794   
33615       64313            1.828985                 8.377309   

           Total Cost(millions)  Total Profit(millions)  
Client ID                                                
24741                 45.688900               36.579992  
38378                  9.634721                3.271830  
66037                  7.004483                3.255032  
46820                  7.007192                2.736603  
33615                  6.175314                2.201995